sklearn/tests/test_common.py

```diff
@@ -296,6 +297,54 @@ def test_estimators_overwrite_params():
                 yield check_estimators_overwrite_params, name, Estimator
 
 
+# These are the known trailing-_ properties on current estimators.
```

Comment:
> The exception is where it would be breaking backwards compatibility to, for example, change an existing error from a ValueError to an AttributeError (although perhaps we could define a custom BackwardsCompatibleAttributeError that extends ValueError and AttributeError!).

This is exactly the purpose of `NotFittedError`. We should probably use `check_is_fitted` [(see here)](https://github.com/scikit-learn/scikit-learn/blob/a2d638b655d06abf9f097dde025024fd27d45895/sklearn/utils/validation.py#L636) everywhere.